presto-main/src/main/java/com/facebook/presto/operator/aggregation/reservoirsample/UnweightedDoubleReservoirSample.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.reservoirsample;
+
+import io.airlift.slice.SizeOf;
+import io.airlift.slice.SliceInput;
+import io.airlift.slice.SliceOutput;
+import io.airlift.slice.Slices;
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.Arrays;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class UnweightedDoubleReservoirSample
+        implements Cloneable
+{
+    private static final int MAX_SAMPLES_LIMIT = 1_000_000;
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(WeightedDoubleReservoirSample.class).instanceSize();
+
+    private int count;
+    private double[] samples;
+
+    public UnweightedDoubleReservoirSample(int maxSamples)
+    {
+        checkArgument(maxSamples > 0, "Number of samples must be positive");
```

Comment:
- "Number of samples" -> "Maximum number of samples" for clarity and consistency
- "is capped" -> "must not exceed " + MAX_SAMPLES_LIMIT to give user the information necessary to fix the query without guessing

```
        checkArgument(maxSamples > 0, "Maximum number of samples must be positive");
        checkArgument(maxSamples <= MAX_SAMPLES_LIMIT, "Maximum number of samples must not exceed " + MAX_SAMPLES_LIMIT);
```